graphql/src/test/java/com/linecorp/armeria/server/graphql/ExecutionIdGeneratorTest.java

```diff
@@ -0,0 +1,130 @@
+            final GraphqlService service =
+                    GraphqlService.builder()
+                                  .configureGraphql(builder -> builder.queryExecutionStrategy(idStrategy))
+                                  .schemaFile(graphqlSchemaFile)
+                                  .runtimeWiring(c -> {
+                                      final DataFetcher<String> bar = dataFetcher("bar");
+                                      c.type("Query",
+                                             typeWiring -> typeWiring.dataFetcher("foo", bar));
+                                  })
+                                  .build();
+            final GraphqlService customizedExecutionIdService =
+                    GraphqlService.builder()
+                                  .executionIdGenerator(new ConcatExecutionIdGenerator())
+                                  .configureGraphql(builder -> builder.queryExecutionStrategy(idStrategy))
+                                  .schemaFile(graphqlSchemaFile)
+                                  .runtimeWiring(c -> {
+                                      final DataFetcher<String> bar = dataFetcher("bar");
+                                      c.type("Query",
+                                             typeWiring -> typeWiring.dataFetcher("foo", bar));
+                                  })
+                                  .build();
+            sb.service("/graphql", service)
+              .service("/graphql-customized-execution-id", customizedExecutionIdService);
+        }
+    };
+
+    static DataFetcher<String> dataFetcher(String value) {
+        return environment -> {
+            final ServiceRequestContext ctx = GraphqlServiceContexts.get(environment);
+            assertThat(ctx.eventLoop().inEventLoop()).isTrue();
+            // Make sure that a ServiceRequestContext is available
+            assertThat(ServiceRequestContext.current()).isSameAs(ctx);
+            return value;
+        };
+    }
+
+    @Test
+    void defaultExecutionIdGenerator() throws InterruptedException {
+        BlockingWebClient.of(server.httpUri()).get("/graphql?query={foo}");
+
+        final ServiceRequestContext ctx = server.requestContextCaptor().take();
+        assertThat(idStrategy.executionId).isEqualTo(ExecutionId.from(ctx.id().text()));
+    }
+
+    @Test
+    void concatenateExecutionIdGenerator() throws InterruptedException {
+        final String operationName = "Foo";
+        final String query = "query " + operationName + " {foo}";
+        final String content = "{\n" +
+                               "  \"operationName\": \"" + operationName + "\",\n" +
+                               "  \"query\": \"" + query + "\"\n" +
+                               "}\n";
+        final HttpRequest request = HttpRequest.builder().post("/graphql-customized-execution-id")
+                                               .content(MediaType.GRAPHQL_RESPONSE_JSON, content)
```

Comment:
`GRAPHQL_RESPONSE_JSON` is only used for responses.
https://github.com/graphql/graphql-over-http/blob/main/spec/GraphQLOverHTTP.md#media-types
```suggestion
                                               .content(MediaType.JSON, content)
```